content/concepts/did-ddo.md

```diff
@@ -34,98 +43,120 @@ In Ocean, a DID is a string that looks like:
+| **`format`**        | `string`              | no          | Packaging format of the software. |
+| **`version`**       | `string`              | no          | Version of the software. |
+| **`container`**     | `Object`              | yes         | Object describing the Docker container image. |
 
-A part of the `attributes.main` sub-section, other optional sub-sections like `attributes.extra` can be added. These depend on the service type.
+The `container` object has the following attributes:
 
-Each service has a `timeout` (in seconds) section describing how long the sevice can be used after consumption is initiated. A timeout of 0 represents no time limit.
+| Attribute           |   Type   | Required  | Description                                         |
+| ------------------- | -------- | --------- | --------------------------------------------------- |
+| **`entrypoint`**    | `string` | yes       | The command to execute, or script to run inside the Docker image. |
+| **`image`**         | `string` | yes       | Name of the Docker image. |
+| **`tag`**           | `string` | yes       | Tag of the Docker image. |
+| **`checksum`**      | `string` | yes       | Checksum of the Docker image. |
 
-The `cost` attribute is obsolete, as of Ocean V3. As of V3, to consume an asset, one sends exactly 1.0 datatokens of the asset, so a `cost` is not needed.
 
-## DDO Service Example
 
-Here is an example DDO service:
 
-```json
-"service": [
-  {
-    "index": 0,
-    "type": "metadata",
-    "serviceEndpoint": "https://service/api/v1/metadata/assets/ddo/did:op:0ebed8226ada17fde24b6bf2b95d27f8f05fcce09139ff5cec31f6d81a7cd2ea",
-    "attributes": {
-      "main": {},
-      "additionalInformation": {},
-      "curation": {}
-    }
-  },
-  {
-    "index": 1,
-    "type": "access",
-    "serviceEndpoint": "http://localhost:8030/api/v1/provider/services/consume",
-    "attributes": {
-      "main": {
-        "cost":"10",
-        "timeout":0
-      },
-      "additionalInformation": {}
-    }
-  },
-  {
-    "index": 2,
-    "type": "compute",
-    "serviceEndpoint": "http://localhost:8030/api/v1/provider/services/compute",
-    "attributes": {
-      "main": {
-        "cost":"10",
-        "timeout":3600
-      },
-      "additionalInformation": {}
-    }
-  }
-]
+
+## Service
+
+| Attribute           | Type                  | Required | Description                                                                                                                                                                                       |
+| ------------------- | --------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **`type`**     | Text          | **Yes**       | Type of service (access, compute, wss, etc |
```

Comment:
we should also add a description here , not required

Reply:
done